src/core/meshcop/border_agent.cpp

```diff
@@ -754,8 +789,14 @@ Error BorderAgent::SetEphemeralKey(const char *aKeyString, uint32_t aTimeout, ui
     Error    error  = kErrorNone;
     uint16_t length = StringLength(aKeyString, kMaxEphemeralKeyLength + 1);
 
-    VerifyOrExit(mState == kStateStarted, error = kErrorInvalidState);
-    VerifyOrExit((length >= kMinEphemeralKeyLength) && (length <= kMaxEphemeralKeyLength), error = kErrorInvalidArgs);
+    VerifyOrExit(mState == kStateStarted, {
+        error = kErrorInvalidState;
+        mCounters.mEpskcInvalidBaStateErrors++;
+    });
+    VerifyOrExit((length >= kMinEphemeralKeyLength) && (length <= kMaxEphemeralKeyLength), {
+        error = kErrorInvalidArgs;
+        mCounters.mEpskcInvalidArgsErrors++;
+    });
```

Comment:
While the use of `{ }` to allow multiple statements in `VerifyOrExit()` is correct, it is harder to read and follow.

How about after the `exit` label, we use a `switch(error)` to update the proper counter?

Reply:
Done